src/OrleansProviders/Streams/Common/Monitors/ICacheMonitor.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Orleans.Providers.Streams.Common
+{
+    /// <summary>
+    /// Responsible for monitoring cache related metrics
+    /// </summary>
+    public interface ICacheMonitor
+    {
+        #region Event driven metrics
+
+        /// <summary>
+        /// Track cache pressure metrics when cache pressure monitor encounter a status change
+        /// </summary>
+        void TrackCachePressureMonitorStatusChange(string pressureMonitorType, bool underPressure, double? cachePressureContributionCount, double? currentPressure);
```

Comment:
but threshold is not a metrics. and the user knows the threshold, they set it. 
It is the same as message age, we are not reporting DataMinAgeInCache as well.